pkg/resourceapply/helpers.go

```diff
@@ -16,6 +16,23 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// ExtractManagedAnnotations extracts managed annotations from the object and returns them.
+func ExtractManagedAnnotations(obj metav1.Object) map[string]string {
+	annotations := obj.GetAnnotations()
+
+	customAnnotations := map[string]string{}
+	for key, value := range annotations {
+		if !strings.HasPrefix(key, "scylla-operator.scylladb.com/") {
```

Comment:
I think this would be a good adept for a separate function (`IsManagedAnnotation`) and also making the prefix a const that is used in the annotation consts so we can be sure those are managed